rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/AbstractBrooklynRestResource.java

```diff
@@ -49,12 +50,18 @@
 
     
     private ManagementContext managementContext;
+    private ManagementContextInternal managementContextInternal;
```

Comment:
After discussing this change still don't see the reason for having both properties. If the resources require `ManagementContextInternal`, then while not change the type of `managementContext` and inject the appropriate type? Or leave the type as is and keep the castings.